src/core_plugins/input_control_vis/public/control/list_control_factory.js

```diff
@@ -35,44 +35,78 @@ const termsAgg = (field, size, direction) => {
 const listControlDelimiter = '$$kbn_delimiter$$';
 
 class ListControl extends Control {
-  constructor(controlParams, filterManager, selectOptions) {
-    super(controlParams, filterManager);
-
-    this.selectOptions = selectOptions;
-  }
 
   getMultiSelectDelimiter() {
     return this.filterManager.delimiter;
   }
+
+  async fetch() {
+    let ancestorSearchSource;
+    if (this.hasAncestors()) {
+      if (this.hasUnsetAncestor()) {
+        this.disable(`Disabled until '${this.ancestors[0].label}' is set.`);
+        // Remove any existing filters for control.
+        this.filterManager.findFilters().forEach((existingFilter) => {
```

Comment:
If you add a parent and child filters, pin both of them and then remove the parent filter via the input control, it's removing the pinned filter.

Reply:
That is the intended behavior. The filter pill is owned by the Control Panel. When the parent is unset, then the child state can only be unknown since there is no way of getting a valid options list

Reply:
Even when I'm manually pinning the pills? It seems unnatural for me to pin a filter, and then have something else remove it. Perhaps this gif will better explain what I'm seeing:

![pinned-filters](https://user-images.githubusercontent.com/627123/35806161-92a4394a-0a4c-11e8-9873-6a6c9ab53bb9.gif)

It's possible that this behavior was present before this PR with non-parent/child pills, but I only noticed it now.